webrender/src/device.rs

```diff
@@ -2215,18 +2307,42 @@ impl<B: hal::Backend> Device<B, hal::Graphics> {
         texture.filter = filter;
         texture.layer_count = layer_count;
         texture.render_target = render_target;
+        //println!("texture.width={:?}", texture.width);
+        //println!("texture.height={:?}", texture.height);
+        //println!("texture.filter={:?}", texture.filter);
+        //println!("texture.layer_count={:?}", texture.layer_count);
+        //println!("texture.render_target={:?}", texture.render_target);
 
         self.bind_texture(DEFAULT_TEXTURE, texture);
         self.set_texture_parameters(/*texture.target,*/ filter);
 
         match render_target {
             Some(info) => {
-                self.update_target_storage(texture, &info, is_resized, pixels);
+                self.update_target_storage(texture, &info, is_resized);
             }
             None => {
-                self.update_image(texture, pixels);
+                self.update_image(texture);
             }
         }
+
+        if let Some(data) = pixels {
+            self.upload_queue
+                .push(
+                    self.images
+                        .get_mut(&texture.id)
+                        .expect("Texture not found.")
```

Comment:
can just do `self.images[&texture.id]` for short, since the error handling doesn't do anything fancy here

Reply:
We need it `mutable`, but `HashMap` does not have `IndexMut`.